widgetssdk/src/main/java/com/glia/widgets/chat/ChatController.java

```diff
@@ -673,56 +710,123 @@ private void changeDeliveredIndex(List<ChatItem> currentChatItems, VisitorMessag
+
+
+                } else {
+                    currentChatItems.add(new OperatorAttachmentItem(
+                            message.getId(),
+                            ChatAdapter.OPERATOR_FILE_VIEW_TYPE,
+                            showChatHead,
+                            file,
+                            chatState.operatorProfileImgUrl, false, false));
+                }
+
+            }
+        }
+    }
+
+    private void addLastMessageItem(List<ChatItem> currentChatItems, ChatMessage message, MessageAttachment attachment) {
+        if (!message.getContent().equals("")) {
             currentChatItems.add(new OperatorMessageItem(
-                    lastItemInView.getId(),
-                    lastItemInView.operatorProfileImgUrl,
-                    false,
-                    lastItemInView.content,
-                    lastItemInView.singleChoiceOptions,
-                    lastItemInView.selectedChoiceIndex,
-                    lastItemInView.choiceCardImageUrl
+                    message.getId(),
+                    null,
+                    true,
+                    message.getContent(),
+                    getSingleChoiceAttachmentOptions(attachment),
+                    null,
+                    getSingleChoiceAttachmentImgUrl(attachment)
             ));
         }
+    }
+
+    private String getSingleChoiceAttachmentImgUrl(MessageAttachment attachment) {
         String imageUrl = null;
+
         if (attachment instanceof SingleChoiceAttachment) {
-            try {
-                imageUrl = ((SingleChoiceAttachment) attachment).getImageUrl().get();
-            } catch (Exception ignored) {
+            Optional<String> optionalImageUrl = ((SingleChoiceAttachment) attachment).getImageUrl();
+
+            if (optionalImageUrl.isPresent()) {
+                imageUrl = optionalImageUrl.get();
             }
         }
 
-        List<SingleChoiceOption> singleChoiceOptions = attachment instanceof SingleChoiceAttachment ?
-                Arrays.asList(((SingleChoiceAttachment) attachment).getOptions()) :
-                null;
-        currentChatItems.add(new OperatorMessageItem(
-                message.getId(),
-                chatState.operatorProfileImgUrl,
-                true,
-                message.getContent(),
-                singleChoiceOptions,
-                null,
-                imageUrl
-        ));
+        return imageUrl;
     }
 
-    private boolean isMessageValid(String message) {
-        return message.length() > 0;
+    private List<SingleChoiceOption> getSingleChoiceAttachmentOptions(MessageAttachment attachment) {
+        List<SingleChoiceOption> singleChoiceOptions = null;
+
+        if (attachment instanceof SingleChoiceAttachment) {
+            singleChoiceOptions = Arrays.asList(((SingleChoiceAttachment) attachment).getOptions());
+        }
+
+        return singleChoiceOptions;
+    }
+
+    private boolean isMessageValid(String message, int numOfAttachment) {
```

Comment:
this one will be not needed after extraction to usecase

Reply:
I think the same as above